Lib/test/test_threading.py

```diff
@@ -656,6 +656,60 @@ def test_BoundedSemaphore_limit(self):
                 t.join()
             self.assertRaises(ValueError, bs.release)
 
+    def test_semaphore_docs(self):
+        """Basic test to validate that the behavior described in the docs is
+        accurate.
+        """
+        s = threading.Semaphore(1)
+
+        # note: no mutex since we know only one thread is executing at once
+        results = []
+
+        def put_result():
+            results.append(s.acquire())
+
+        th1 = threading.Thread(target=put_result)
+        th2 = threading.Thread(target=put_result)
+        th3 = threading.Thread(target=put_result)
+
+        th1.start()
+        th2.start()
+        th3.start()
+
+        start = time.time()
+        while th1.is_alive():
+            assert time.time() - start < 0.5
```

Comment:
Please use unittest methods like `self.assertEqual()` and family: https://docs.python.org/3/library/unittest.html#assert-methods

Reply:
There are tests for semaphores in `BaseSemaphoreTests` in `Lib/test/lock_tests.py`. You should look there and add any tests there as needed instead.

Reply:
(but everything seems to be tested already there, except the return value of blocking semaphores which wasn't really documented anyway)

Reply:
I didn't see those, thanks for pointing it out.

I've removed the "doc tests" and added testing the return type to the base semaphore tests. Let me know if that looks good!